File: Data Structures & Algorithms/trapping-rain-water/submission-0.js
class Solution {
    /**
     * @param {number[]} height
     * @return {number}
     */
    trap(height) {
        let area = 0;
        let maxLeftHeight = 0;
        let maxRightHeight = 0;
        let left = 0;
        let right = height.length - 1;

        while (left < right) {
            maxLeftHeight = Math.max(maxLeftHeight, height[left]);
            maxRightHeight = Math.max(maxRightHeight, height[right]);

            if (height[left] < height[right]) {
                area += maxLeftHeight - height[left];
                left++;
            } else {
                area += maxRightHeight - height[right];
                right--;
            }
        }

        return area;
    }
}

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-0.py
class Solution:
    def lengthOfLongestSubstring(self, s: str) -> int:
        if len(s) <= 1:
            return len(s)

        left = 0
        longest = 1
        seen = set()

        for right, curr in enumerate(s):
            while s[right] in seen:
                seen.remove(s[left])
                left += 1

            seen.add(s[right])
            longest = max(longest, right - left + 1)
        
        return longest